.circleci/config.yml

```diff
@@ -1,42 +1,118 @@
 version: 2.1
 
-jobs:
-  standards:
+commands:
+  install-dependencies:
+    description: "Install development dependencies."
+    steps:
+      - run: composer install
+      - run: npm ci
+  mkdir_artifacts:
+    description: "Make Artifacts directory"
+    steps:
+      - run:
+          command: |
+            [ ! -d "/tmp/artifacts" ] && mkdir /tmp/artifacts &>/dev/null
+
+executors:
+  base:
+    docker:
+      - image: circleci/buildpack-deps:latest
+    working_directory: /tmp
+  php_node:
     docker:
       - image: circleci/php:7.3.3-stretch-node-browsers
+    working_directory: /tmp/genesis-sample
+  python:
+    docker:
+      - image: circleci/python:3.7-stretch
+    working_directory: /tmp
+
+jobs:
+  checkout:
+    executor: base
     steps:
-      - checkout
+      - checkout:
+          path: genesis-sample
+      - persist_to_workspace:
+          root: /tmp
+          paths:
+            - genesis-sample
+
+  standards:
+    executor: php_node
+    steps:
+      - attach_workspace:
+          at: /tmp
       - install-dependencies
       - run: composer phpcs
       - run: npm run lint:css
       - run: npm run lint:js
 
   bundle:
-    docker:
-      - image: circleci/node:12.12.0
+    executor: php_node
     steps:
-      - checkout
+      - attach_workspace:
+          at: /tmp
       - run: npm ci
       - run: npm run zip
-      - run: mkdir /tmp/artifacts
+      - mkdir_artifacts
       - run: mv ./*.zip /tmp/artifacts/.
+      - persist_to_workspace:
+          root: /tmp
+          paths:
+            - artifacts
+            - genesis-sample
       - store_artifacts:
           path: /tmp/artifacts/
 
-commands:
-  install-dependencies:
-    description: "Install development dependencies."
+  create_data_file:
+    executor: php_node
     steps:
-      - run: composer install
-      - run: npm ci
+      - attach_workspace:
+          at: /tmp
+      - run: ls -lash
+
+  deploy_s3:
+    executor: python
+    steps:
+      - attach_workspace:
+          at: /tmp
+      - aws-s3/sync:
+          from: "/tmp/artifacts/"
+          to: "s3://<need_bucket_url>"
+
 workflows:
   version: 2
   check-standards:
     jobs:
-      - standards
+      - checkout
+      - standards:
+          requires:
+            - checkout
       - bundle:
+          requires:
+            - standards
```

Comment:
The jobs that this job is dependent on might not run on a tag by default.  You will probably have to add the tags: filter to 'checkout' and 'standards' also.

Reply:
Noted for the next PR I'll be making

Reply:
Why not fix in this PR?  Otherwise I'll lose context of where I asked the question and gave PR feedback/review and where the fix/change was made.  Otherwise, maybe non of this should be in this PR at all, if it's outside the scope of the original work for this PR.

Reply:
@BMO-tech +1 to Marky Marks sentiment of adding to current PR for posterity sake. Our future selves with thank you for doing so.

Reply:
Should I have just deleted the draft PR and created a new one? Changes in `.circleci/config.yml` were not part of this PR's scope. I had accidentally created a new branch from my previous work instead of from `develop`.
I fixed this mix up before marking the draft as ready to review, so the current PR does not have any `.circleci/config.yml` changes at all.

Reply:
Definitely don't want to be prescriptive in the how per se. Thinking more along the lines of making sure its consumable in the future as we come back to it, or some stranger we don't even know yet does. _waves to future strangers_